core/src/main/java/org/rfcx/guardian/utility/device/DeviceCPU.java

```diff
@@ -69,29 +69,31 @@ private void incrementClockSpeedAverage() {
 	}
 	
 	private static float getCurrentCPUPercentage(String logTag) {
-		try {
-	        RandomAccessFile reader = new RandomAccessFile("/proc/stat", "r");
-	        String load = reader.readLine();
-	        String[] toks = load.split(" ");
-	        long idle1 = Long.parseLong(toks[5]);
-	        long cpu1 = Long.parseLong(toks[2]) + Long.parseLong(toks[3]) + Long.parseLong(toks[4])
-	              + Long.parseLong(toks[6]) + Long.parseLong(toks[7]) + Long.parseLong(toks[8]);
-	        try {
-	        		SystemClock.sleep(SAMPLE_DURATION_MILLISECONDS);
-	        } catch (Exception e) { 
-	        		RfcxLog.logExc(logTag, e); 
-	        }
-	        reader.seek(0);
-	        load = reader.readLine();
-	        reader.close();
-	        toks = load.split(" ");
-	        long idle2 = Long.parseLong(toks[5]);
-	        long cpu2 = Long.parseLong(toks[2]) + Long.parseLong(toks[3]) + Long.parseLong(toks[4])
-	            + Long.parseLong(toks[6]) + Long.parseLong(toks[7]) + Long.parseLong(toks[8]);
-	        return (float)(cpu2 - cpu1) / ((cpu2 + idle2) - (cpu1 + idle1));
-	    } catch (IOException e) {
-	    		RfcxLog.logExc(logTag, e);
-	    }
+		if (DeviceRoot.isRooted()) {
+			try {
+				RandomAccessFile reader = new RandomAccessFile("/proc/stat", "r");
+				String load = reader.readLine();
+				String[] toks = load.split(" ");
+				long idle1 = Long.parseLong(toks[5]);
+				long cpu1 = Long.parseLong(toks[2]) + Long.parseLong(toks[3]) + Long.parseLong(toks[4])
+						+ Long.parseLong(toks[6]) + Long.parseLong(toks[7]) + Long.parseLong(toks[8]);
+				try {
+					SystemClock.sleep(SAMPLE_DURATION_MILLISECONDS);
+				} catch (Exception e) {
+					RfcxLog.logExc(logTag, e);
+				}
+				reader.seek(0);
+				load = reader.readLine();
+				reader.close();
+				toks = load.split(" ");
+				long idle2 = Long.parseLong(toks[5]);
+				long cpu2 = Long.parseLong(toks[2]) + Long.parseLong(toks[3]) + Long.parseLong(toks[4])
+						+ Long.parseLong(toks[6]) + Long.parseLong(toks[7]) + Long.parseLong(toks[8]);
+				return (float) (cpu2 - cpu1) / ((cpu2 + idle2) - (cpu1 + idle1));
+			} catch (IOException e) {
+				RfcxLog.logExc(logTag, e);
+			}
+		}
```

Comment:
These lines have not changed... try not to commit them.

Use the technique I suggested before:

```
if (!DeviceRoot.isRooted()) {
   return 0;
}
```